### ISSUE TITLE
Highlighting lost based on cursor position

### ISSUE THREAD

**Issue by AlanWarren** [styled-components/vim-styled-components#49]
Hello, 

When editing large components with many nested styled component definitions, i will often see everything below my styled component definitions highlight as jsTemplateString until I move the cursor back within a styled component's definition.  

I'm using a fairly default setup, with pangloss JS highlighting and vim-jsx-improve.  

Few pics with different cursor positions. 
![screen shot 2018-08-22 at 10 38 32 am](https://user-images.githubusercontent.com/294606/44474266-e1e22700-a5f7-11e8-8be9-473c686d975d.png)
![screen shot 2018-08-22 at 10 38 54 am](https://user-images.githubusercontent.com/294606/44474285-ed355280-a5f7-11e8-85ec-9b2e1eeeea62.png)

Thanks!

**Comment by AlanWarren** [comment]
Sorry, I forgot to include this.  I've tossed together a minimal init.vim.  The above screenshots use my own colorscheme, but I tried a few popular ones I found on google, and they did the same thing. 

`syntax on
call plug#begin('~/.config/nvim/plugged')
Plug 'pangloss/vim-javascript'
Plug 'mxw/vim-jsx'
Plug 'styled-components/vim-styled-components', { 'branch': 'main' }
call plug#end()

colorscheme PaperColor-Dark
set termguicolors
set t_Co=256
set t_ut=
` 
I have some experience with hacking at syntax files,  and I'm willing to put some effort towards this, but I'm honestly a bit lost due to the cursor issue.  I'm not sure why the highlighting cares where my mouse is.  ¯\_(ツ)_/¯

ty sir
